src/analysis/normed_space/quaternion_exponential.lean

```diff
@@ -0,0 +1,137 @@
+
+lemma conj_exp (q : ℍ[ℝ]) : conj (exp ℝ q) = exp ℝ (conj q) := star_exp q
+
+@[simp, norm_cast] lemma exp_coe (r : ℝ) : exp ℝ (r : ℍ[ℝ]) = ↑(exp ℝ r) :=
+(map_exp ℝ (algebra_map ℝ ℍ[ℝ]) (continuous_algebra_map _ _) _).symm
+
+/-- Auxiliary result; if the power series corresponding to `real.cos` and `real.sin` evaluatated
+at `‖q‖` tend to `c` and `s`, then the exponential series tends to `c + (s / ‖q‖)`. -/
+lemma has_sum_exp_series_of_imaginary
+  {q : quaternion ℝ} (hq : q.re = 0) {c s : ℝ}
+  (hc : has_sum (λ n, (-1)^n * ‖q‖^(2 * n) / (2 * n)!) c)
+  (hs : has_sum (λ n, (-1)^n * ‖q‖^(2 * n + 1) / (2 * n + 1)!) s) :
+  has_sum (λ n, exp_series ℝ _ n (λ _, q)) (↑c + (s / ‖q‖) • q) :=
+begin
+  replace hc := has_sum_coe.mpr hc,
+  replace hs := (hs.div_const ‖q‖).smul_const q,
+  obtain rfl | hq0 := eq_or_ne q 0,
+  { simp_rw [exp_series_apply_zero, norm_zero, div_zero, zero_smul, add_zero],
+    simp_rw [norm_zero] at hc,
+    convert hc,
+    ext (_ | n) : 1,
+    { rw [pow_zero, mul_zero, pow_zero, nat.factorial_zero, nat.cast_one, div_one, one_mul,
+        pi.single_eq_same, coe_one], },
+    { rw [zero_pow (mul_pos two_pos (nat.succ_pos _)), mul_zero, zero_div,
+        pi.single_eq_of_ne (n.succ_ne_zero), coe_zero], } },
+  simp_rw exp_series_apply_eq,
+  have hq2 : q^2 = -norm_sq q := sq_eq_neg_norm_sq.mpr hq,
+  have hqn := norm_ne_zero_iff.mpr hq0,
+  refine has_sum.even_add_odd _ _,
+  { convert hc using 1,
+    ext n : 1,
+    let k : ℝ := ↑(2 * n)!,
+    calc k⁻¹ • q ^ (2 * n)
+        = k⁻¹ • ((-norm_sq q) ^ n) : by rw [pow_mul, hq2]
+    ... = k⁻¹ • ↑((-1) ^ n * ‖q‖ ^ (2 * n)) : _
+    ... = ↑((-1) ^ n * ‖q‖ ^ (2 * n) / k) : _,
+    { congr' 1,
+      rw [neg_pow, norm_sq_eq_norm_sq, pow_mul, sq],
+      push_cast },
+    { rw [←coe_mul_eq_smul, div_eq_mul_inv],
+      norm_cast,
+      ring_nf } },
+  { convert hs using 1,
+    ext n : 1,
+    let k : ℝ := ↑(2 * n + 1)!,
+    calc k⁻¹ • q ^ (2 * n + 1)
+        = k⁻¹ • ((-norm_sq q) ^ n * q) : by rw [pow_succ', pow_mul, hq2]
+    ... = k⁻¹ • ((-1) ^ n * ‖q‖ ^ (2 * n)) • q : _
+    ... = ((-1) ^ n * ‖q‖ ^ (2 * n + 1) / k / ‖q‖) • q : _,
+    { congr' 1,
+      rw [neg_pow, norm_sq_eq_norm_sq, pow_mul, sq, ←coe_mul_eq_smul],
+      push_cast },
+    { rw smul_smul,
+      congr' 1,
+      simp_rw [pow_succ', mul_div_assoc, div_div_cancel_left' _ _ hqn],
+      ring } },
+end
+
+/-- The closed for for the quaternion exponential on imaginary quaternions. -/
+lemma exp_of_re_eq_zero (q : quaternion ℝ) (hq : q.re = 0) :
+  exp ℝ q = ↑(real.cos ‖q‖) + (real.sin ‖q‖ / ‖q‖) • q :=
+begin
+  rw [exp_eq_tsum],
+  refine has_sum.tsum_eq _,
+  simp_rw ← exp_series_apply_eq,
+  exact has_sum_exp_series_of_imaginary hq (real.has_sum_cos _) (real.has_sum_sin _),
+end
+
+/-- The closed form for the quaternion exponential on arbitrary quaternions. -/
+lemma exp_eq (q : quaternion ℝ) :
+  exp ℝ q = exp ℝ q.re • (
+    let v := q - q.re in ↑(real.cos ‖v‖) + (real.sin ‖v‖ / ‖v‖) • v) :=
```

Comment:
Having a `let` binding in a lemma statement doesn't sound like a great idea. Do you want to define `q.im := q - q.re`? Then you could state skew-adjointness as `q = q.im` (analogously to `q = q.re`).

Reply:
I was reluctant because it's lots of boilerplate. Does #18456 look worth it to you?

Reply:
I quite like it, I must say!

Reply:
I've updates this PR to depend on that change.